packages/teams-js/src/internal/communication.ts

```diff
@@ -47,15 +53,18 @@ class CommunicationPrivate {
   public static topMessageQueue: MessageRequest[] = [];
   public static nextMessageId = 0;
   public static callbacks: {
-    [id: number]: Function; // (arg1, arg2, ...etc) => void
+    [id: MessageID]: Function; // (arg1, arg2, ...etc) => void
```

Comment:
>MessageID [](http://example.com/codeflow?start=9&length=9)

It looks like these maps are still using a `string` (which is the serialized format for the UUID) rather than an actual UUID type. Can we use an actual UUID type for our maps, rather than relying on the implementation detail of how a UUID is serialized?

Reply:
Sorry, I'm a bit confused by this question. The ids are using the type `MessageID`. Unfortunately, I can't use an object as an index and have to use either a template literal, number or string.

Reply:
Is there a reason you are restricted to using this data structure to store the callbacks? You could consider using a JavaScript `Map` (although I'm guessing that won't work because you probably want custom comparison semantics on your keys). It seems like using the [Map structure in the immutable-js library](https://github.com/immutable-js/immutable-js/#equality-treats-collections-as-values) would suit this application.